src/changes/updateChangesStatus.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * © 2023 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { closeHistory } from 'prosemirror-history'
+import { EditorState, Transaction } from 'prosemirror-state'
+
+import { ChangeSet } from '../ChangeSet'
+import { CHANGE_OPERATION, CHANGE_STATUS, TextChange, TrackedChange } from '../types/change'
+import { applyAcceptedRejectedChanges } from './applyChanges'
+import { updateChangeAttrs } from './updateChangeAttrs'
+
+export function updateChangesStatus(
+  createdTr: Transaction,
+  changeSet: ChangeSet,
+  ids: string[],
+  status: CHANGE_STATUS,
+  userID: string,
+  oldState: EditorState
+) {
+  const change = changeSet.get(ids[0])
+  if (change && status !== CHANGE_STATUS.pending) {
+    const textChanges: TextChange[] = []
+    const nonTextChanges: TrackedChange[] = []
+
+    changeSet.changes.forEach((c) => {
+      if (ids.includes(c.id)) {
+        c.dataTracked.status = status
+        if (ChangeSet.isTextChange(c)) {
+          textChanges.push(c)
+        } else {
+          nonTextChanges.push(c)
```

Comment:
will need to add:
```
if (change.dataTracked.operation === CHANGE_OPERATION.node_split) {
            nonTextChanges.push(changeSet.changes.find(
              (c) => c.dataTracked.operation === 'reference' && c.dataTracked.referenceId === change.id
            )!)
          }
```

Reply:
Why? My understanding that "reference" change is not an applicable change and we need to remove it when we reject a change. Which is what we do in this block - https://github.com/Atypon-OpenSource/manuscripts-track-changes-plugin/pull/32/files#diff-59d17d18bdd3457b78411a4684d1059c99545d78fff14bec2b21e1dc18512ba6R55 


Do you mean that we need to revert that we need to drop that change when we accept a related split and there we need to include that?

Reply:
yes we need to drop that for both accepted/rejected. and in the link you point we handle just `node_split`

Reply:
updated

Reply:
can you please move it after `nonTextChanges.push(c)`

Reply:
I moved it, but doesnt make any difference. Why did you want it to be moved there?

Reply:
> Why did you want it to be moved there?

to use reference first in the revert, and then after that will drop it

Reply:
the function that does reverting sorts all the changes in chronological order anyway so that doesn't affect the flow (see applyAcceptedRejectedChanges)